### ISSUE TITLE
Issue enabling tags

### ISSUE THREAD

**Issue by SilentGlasses** [squidfunk/mkdocs-material#4150]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Enabling tags feature breaks site with the following error:
```
Error reading page 'tags.md': '<' not supported between instances of 'NoneType' and 'str'
```

### Expected behaviour

Site should build just fine.

### Actual behaviour

After enabling the tags feature, my site refuses to load...  Getting this output.

```
mkdocs serve
INFO     -  Building documentation...
INFO     -  Cleaning site directory
ERROR    -  Error reading page 'tags.md': '<' not supported between instances of 'NoneType' and 'str'
Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/mkdocs/__main__.py", line 181, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, watch=watch, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 63, in serve
    config = builder()
  File "/usr/local/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 58, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/usr/local/lib/python3.10/site-packages/mkdocs/commands/build.py", line 292, in build
    _populate_page(file.page, config, files, dirty)
  File "/usr/local/lib/python3.10/site-packages/mkdocs/commands/build.py", line 170, in _populate_page
    page.markdown = config['plugins'].run_event(
  File "/usr/local/lib/python3.10/site-packages/mkdocs/plugins.py", line 102, in run_event
    result = method(item, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/material/plugins/tags/plugin.py", line 77, in on_page_markdown
    return self.__render_tag_index(markdown)
  File "/usr/local/lib/python3.10/site-packages/material/plugins/tags/plugin.py", line 101, in __render_tag_index
    for args in sorted(self.tags.items())
TypeError: '<' not supported between instances of 'NoneType' and 'str'
```

The only thing on the `tags.md` page is 

```
# Tags

Following is a list of relevant tags:

[TAGS]
```

### Steps to reproduce

1. Enable the tags plugin:
```
plugins:
  - tags:
      tags_file: tags.md
```
2. Add the tags.md page in the docs directory
3. run `mkdocs serve` and get errors

### Package versions

- Python: `python --version`
  - Python 3.10.5
- MkDocs: `mkdocs --version`
  - mkdocs, version 1.3.0 from /usr/local/lib/python3.10/site-packages/mkdocs (Python 3.10)
- Material: `pip show mkdocs-material | grep -E ^Version`
  - Version: 8.3.9

### Configuration

```yaml
# Project information
site_name: TechOps Home
site_description: Home of TechOps on Git
site_url: https://pages.git.enova.com/ops/techops_docs/
repo_url: https://git.enova.com/ops/techops_docs
repo_name: TechOps Docs Repo

nav:
  - Home: index.md
  - Tags: tags.md

# ↓↓ ---------------- DO NOT EDIT ANYTHING BELOW THIS LINE ---------------- ↓↓ #
# ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓ #

theme:
  name: 'material'
  custom_dir: docs/overrides
  features:
    - announce.dismiss
    - content.code.annotate
    - navigation.indexes
    - navigation.instant
    - navigation.tabs
    - navigation.tabs.sticky
    - navigation.top
    - search.highlight
    - search.suggest
    - search.share
    - toc.integrate

  logo: 'images/techops_icon.png'
  favicon: 'images/siteicon.png'
  font:
    text: 'IBM Plex Sans'
    code: 'IBM Plex Mono'
  palette:
    primary: 'blue'
    accent: 'deep orange'
  icon:
    repo: fontawesome/brands/github
extra_css:
  - 'stylesheets/extra.css'
extra_javascript:
  - javascripts/mathjax-config.js
  - https://polyfill.io/v3/polyfill.min.js?features=es6
  - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
  - https://cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/loader.js
  - https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.js

# Plugins and Extensions
plugins:
  - search
  - autolinks
  - tags:
      tags_file: tags.md
  - git-revision-date-localized:
      type: timeago
      locale: en
      fallback_to_build_date: true
      exclude:
        - index.md
markdown_extensions:
  - abbr
  - admonition
  - attr_list
  - codehilite:
      guess_lang: true
      use_pygments: true
  - def_list
  - footnotes
#<!----------- DO NOT ENABLE THESE EXTENSIONS - THEY BREAK THINGS ------------->
  #- markdown_include.include:
  #    base_path: .
  #- fontawesome_markdown
#<!---------------------------------------------------------------------------->
  - md_in_html
  - meta
  - pymdownx.arithmatex:
      generic: true
  - pymdownx.betterem:
      smart_enable: all
  - pymdownx.caret
  - pymdownx.mark
  - pymdownx.tilde
  - pymdownx.critic
  - pymdownx.details
  - pymdownx.emoji:
      emoji_index: !!python/name:materialx.emoji.twemoji
      emoji_generator: !!python/name:materialx.emoji.to_svg
  - pymdownx.highlight
  - pymdownx.inlinehilite
  - pymdownx.keys
  - pymdownx.magiclink:
      repo_url_shorthand: true
      user: squidfunk
      repo: mkdocs-material
  - pymdownx.pathconverter
  - pymdownx.saneheaders
  - pymdownx.smartsymbols
  - pymdownx.snippets:
      base_path: .
      check_paths: true
  - pymdownx.superfences:
      custom_fences:
        - name: mermaid
          class: mermaid
          format: !!python/name:pymdownx.superfences.fence_code_format
          format: !!python/name:pymdownx.superfences.fence_div_format
  - pymdownx.tabbed:
      alternate_style: true
  - pymdownx.tasklist:
      custom_checkbox: true
      clickable_checkbox: true
  - toc:
      permalink: false
      toc_depth: 4
```

### System information

- Operating system: macOS Monterey
- Browser: Firefox and Chrome (latest)

**Comment by SilentGlasses** [comment]
```
pip list
Package                                   Version
----------------------------------------- -------
Babel                                     2.10.3
click                                     8.1.3
fontawesome-markdown                      0.5.0
ghp-import                                2.1.0
gitdb                                     4.0.9
GitPython                                 3.1.27
importlib-metadata                        4.12.0
Jinja2                                    3.1.2
livereload                                2.6.3
lunr                                      0.6.2
Markdown                                  3.4.1
markdown-include                          0.6.0
MarkupSafe                                2.1.1
mercurial                                 6.1.4
mergedeep                                 1.3.4
mkdocs                                    1.3.0
mkdocs-autolinks-plugin                   0.6.0
mkdocs-git-revision-date-localized-plugin 1.1.0
mkdocs-material                           8.3.9
mkdocs-material-extensions                1.0.3
packaging                                 21.3
pip                                       22.1.2
pycurl                                    7.45.1
Pygments                                  2.12.0
pymdown-extensions                        9.5
pyparsing                                 3.0.9
python-dateutil                           2.8.2
pytz                                      2022.1
PyYAML                                    6.0
pyyaml_env_tag                            0.1
setuptools                                62.6.0
six                                       1.16.0
smmap                                     5.0.0
tabulate                                  0.8.10
tornado                                   6.2
watchdog                                  2.1.9
wheel                                     0.37.1
zipp                                      3.8.1
```

**Comment by SilentGlasses** [comment]
Not sure how I missed this but... https://github.com/squidfunk/mkdocs-material/issues/4054

It seems like removing the `tags_file: tags.md` fixes things but I want to have a tags page so users can check for things that fall under a certain tag as well as search.

**Comment by SilentGlasses** [comment]
I do notice that on teh page, it's rendering like so:

```
<h1 id="tags">Tags</h1>
<p>Following is a list of relevant tags:</p>
<p>[TAGS]</p>

  <hr>
<div class="md-source-file">
  <small>
    
      Last update:
      <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago"><span class="timeago" datetime="2021-06-22T16:14:14+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date">2021-06-22</span>
      
    
  </small>
</div>
```

Could it be the  extra revision date that is causing this?  If yes, is there a way to ignore extra content on the page?

**Comment by SilentGlasses** [comment]
Actually, found how to exclude the `tags.md` page but it still fails

```
plugins:
  - search
  - tags:
      tags_file: tags.md
  - git-revision-date-localized:
      type: timeago
      locale: en
      fallback_to_build_date: true
      exclude:
        - index.md
        - tags.md
```

**Comment by SilentGlasses** [comment]
Hmm, not sure why the existing issues did not show up..  Found this as well https://github.com/squidfunk/mkdocs-material/issues/3628..

I checked tho and it seems that all the pages so far that have a metadata section are not missing any, however, there are spacing differences as well as some pages without the metadata section, does that matter or do all pages need it?

Doe the spacing matter? Is it a must that there are two spaces?

```
---
tags:
 - tag1
 - tag2
---
```
